README.md

```diff
@@ -1,12 +1,103 @@
-# Git Repository Template
+# Survey Api Sdk Android
 
-Project repository template to set up all public projects at [Nimble](https://nimblehq.co/)
+[![](https://jitpack.io/v/nimblehq/survey-api-sdk-android.svg)](https://jitpack.io/#nimblehq/survey-api-sdk-android)
+
+An Android SDK that support the interactions with our Survey-API, allowing the users to get all surveys `/surveys` or get a survey by id `/surveys/{id}`.
+The SDK is also able to to handle all requests for an API without redefine HTTP framework (like `Retrofit`, `OkHttp`,ect...).
+
+## Installatation
+
+Step 1. Add the JitPack repository to your build file
+
+````
+
+allprojects {
+    repositories {
+        ...
+        maven { url "https://jitpack.io" }
+    }
+}
+
+````
+
+Step 2. Add the dependency
+
+````
+
+dependencies {
+    implementation 'com.github.nimblehq:survey-api-sdk-android:[LATEST_VERSION]'
+}
+
+````
 
 ## Usage
 
-Clone the repository
+- Setup the SDK inside `onCreate` function of the `Application` layer
+
+````
+
+SurveyApi.instance
+            .withMode(BuildConfig.DEBUG) // If debug, it will have log on http request (Optional)
+            .withBaseUrl(BuildConfig.BASE_URL) // Base url to call the request
+            .withCredentials(// Setup credentials information
+                Credentials(
+                    clientID = BuildConfig.CLIENT_ID,
+                    clientSecret = BuildConfig.CLIENT_SECRET
+                )
+            )
+            
+ ````
+
+- Update token to SDK after having it (by Athentication step, for example)
+
+````
+
+with(SurveyApi.instance) {
+      setTokenKey(login.tokenKey)
+      setTokenType(login.tokenType)
+}
+
+````
+
+- Call request for getting SurveyList / SurveyDetail (The SDK already perform on background, it will return the data callback on UI thread)
+
+````
+
+SurveyApi.instance.getSurveyList(page = 1, size = 10, onResponse = {
+            // Handle it as the result
+        })
+        
+````
+
+````
+
+SurveyApi.instance.getSurveyDetail(surveyId = id, onResponse = {
+            // Handle it as the result
+        })
+        
+````
+        
+ - If you want to use your own API, just call:
+
+````
+
+val authService = SurveyApi.instance.buildService<AuthService>()
+
+````
```

Comment:
[This convention](https://nimblehq.co/compass/intro/contribute/#markdown) is being used for compass contribution, and it could be applied for all markdown documents in our development process too. It says we should `Use correct indent for list...`
```suggestion
 - If you want to use your own API, just call:

    ````
    val authService = SurveyApi.instance.buildService<AuthService>()
    ````
```